function-aws/src/main/java/io/micronaut/function/aws/MicronautRequestHandler.java

```diff
@@ -43,19 +43,24 @@
     @SuppressWarnings("unchecked")
     private final Class<I> inputType = initTypeArgument();
 
+    private final MicronautRequestHandler<I, O> injectedInstance;
+
+    public MicronautRequestHandler() {
+        buildApplicationContext(null);
+        injectedInstance = applicationContext.inject(this);
+    }
+
     @Override
     public final O handleRequest(I input, Context context) {
-        if (applicationContext == null) {
-            applicationContext = buildApplicationContext(context);
-        }
+
         if (context != null) {
             registerContextBeans(context, applicationContext);
         }
 
         if (!inputType.isInstance(input)) {
             input = convertInput(input);
         }
-        return applicationContext.inject(this).execute(input);
+        return injectedInstance.execute(input);
```

Comment:
Only change is I don't think the extra field and `injectedInstance` is necessary, since you can just refer to `this`

Reply:
Done!